src/nbl/video/CCUDASharedSemaphore.cpp

```diff
@@ -0,0 +1,18 @@
+// Copyright (C) 2018-2020 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#include "nbl/video/CCUDADevice.h"
+
+#ifdef _NBL_COMPILE_WITH_CUDA_
+namespace nbl::video
+{
+CCUDASharedSemaphore::~CCUDASharedSemaphore()
+{
+	auto& cu = m_device->getHandler()->getCUDAFunctionTable();
+	cu.pcuDestroyExternalSemaphore(m_handle);
+	CloseHandle(m_osHandle);
```

Comment:
if `CloseHandle` is only supposed to be called once after all different API objects acrossmultiple APIs are destroyed, then we need to  have reference counted handles (i.e. `std::shared_ptr`), NOT have the CloseHandle live inside a destructor!